contentcuration/contentcuration/models.py

```diff
@@ -1983,6 +1984,19 @@ class Meta:
         ]
 
 
+class Embeddings(models.Model):
+    """
+    A model that caches embeddings.
+    """
+    content_id = UUIDField(primary_key=False)
+    embedding = VectorField()
```

Comment:
1. We'll also want something on here to track the specific model/version used to generate each embedding.

2. And as we have some cases (e.g. Khan Academy) where content_id is duplicated across languages, we'll need to track the language on here as well. For the Kaggle dataset, what I did was compute an ID I called a "cid", which was a hash of the content_id and language -- so that could be an option also (or just appending, i.e. "{content_id}-{lang_code}" to make a composite ID).

3. We'll then want an efficient way to map from those IDs back into the canonical content node that will be used for the recommendation itself. I'd suggest something like:

```
class ContentNodeLookup(models.Model):
    cid = models.CharField(primary_key=True)
    contentnode = fields.ForeignKey(ContentNode)
```
and then the `Embeddings` class can foreignkey onto `ContentNodeLookup`.

Reply:
@jamalex your points are really important considerations. We'll need to refactor our model. I've some questions on my mind --

- I want to know for what all contentnodes we will be generating embeddings? All public channel contentnodes? 

- When will embeddings be generated? When they are made public, or may be when they are published?

- With each new model release, we will be generating a new set of embeddings, right? For e.g. we have generated embeddings for 100 contentnodes with model `cool-model-v1`, now when we'll have `cool-model-v2` available we'll generate embeddings for those 100 contentnodes again, right?

Reply:
Hi @vkWeb !

> I want to know for what all contentnodes we will be generating embeddings? All public channel contentnodes?

I think this is for public channel for now but should be extendable to other channels

> When will embeddings be generated? When contentnodes are made public, or may be when contentnodes are published?

This will be when the contentnodes are published and also when an update has been made to the channel(or content nodes for that matter)

> With each new model release, we will be generating a new set of embeddings, right? For e.g. we have generated embeddings for 100 contentnodes with model cool-model-v1, now when we'll have cool-model-v2 available we'll generate embeddings for those 100 contentnodes again, right?

I believe your understanding is correct! The recommendations system will rely on the previous model until the new one is up and running. Also just noting here that it most likely will be an ensemble  of models and not necessarily one.
 
 Tagging @jamalex and @bjester incase you have any deferring thoughts to the above.

Reply:
@akolson @jamalex updates to the model landing soon.

Reply:
Updated PR to reflect new structure of models. cc @jamalex @akolson @bjester.